bugsnag-android-core/src/main/java/com/bugsnag/android/DeliveryDelegate.java

```diff
@@ -55,7 +56,11 @@ void deliver(@NonNull Event event) {
             String severityReasonType = event.getImpl().getSeverityReasonType();
             boolean promiseRejection = REASON_PROMISE_REJECTION.equals(severityReasonType);
             boolean anr = event.getImpl().isAnr(event);
-            cacheEvent(event, anr || promiseRejection);
+            if (immutableConfig.getAttemptDeliveryOnCrash()) {
+                cacheAndSendSynchronously(event);
```

Comment:
`DeliveryDelegate.deliver` gets called for all kinds of higher level errors, including ANRs and handled errors (via `Client.populateAndNotifyAndroidEvent`). I don't think we should be blocking the thread for ANRs and non-fatal events.

Reply:
I agree on ANRs and such. I'll modify the order of these `if` checks to only act on unhandled errors that were not previously `attemptSend`.